Cannot build in Windows

Here is the output from attempting a `pip` install. Similar outputs are shown using the other provided methods.

```
C:\windows\system32>pip install miniupnpc
Collecting miniupnpc
c:\python27\lib\site-packages\pip\_vendor\requests\packages\urllib3\util\ssl_.py
:90: InsecurePlatformWarning: A true SSLContext object is not available. This pr
events urllib3 from configuring SSL appropriately and may cause certain SSL conn
ections to fail. For more information, see https://urllib3.readthedocs.org/en/la
test/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Using cached miniupnpc-1.9.tar.gz
Building wheels for collected packages: miniupnpc
  Running setup.py bdist_wheel for miniupnpc
  Complete output from command c:\python27\python.exe -c "import setuptools;__fi
le__='c:\\users\\gappleto\\appdata\\local\\temp\\pip-build-wq5lzh\\miniupnpc\\se
tup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exe
c'))" bdist_wheel -d c:\users\gappleto\appdata\local\temp\tmpezh6hspip-wheel-:
  cc -fPIC -O -Wall -W -Wstrict-prototypes -fno-common -DMINIUPNPC_SET_SOCKET_TI
MEOUT -DMINIUPNPC_GET_SRC_ADDR -D_BSD_SOURCE -D_POSIX_C_SOURCE=1 -ansi   -c -o u
pnpc.o upnpc.c
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, uname -s, ...) failed.
  process_begin: CreateProcess(NULL, cc -fPIC -O -Wall -W -Wstrict-prototypes -f
no-common -DMINIUPNPC_SET_SOCKET_TIMEOUT -DMINIUPNPC_GET_SRC_ADDR -D_BSD_SOURCE
-D_POSIX_C_SOURCE=1 -ansi -c -o upnpc.o upnpc.c, ...) failed.
  make (e=2): The system cannot find the file specified.
  make: *** [upnpc.o] Error 2
  running bdist_wheel
  running build
  running build_ext
  building 'miniupnpc' extension
  creating build
  creating build\temp.win32-2.7
  creating build\temp.win32-2.7\Release
  C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for Pytho
n\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include -
Ic:\python27\PC /Tcminiupnpcmodule.c /Fobuild\temp.win32-2.7\Release\miniupnpcmo
dule.obj
  miniupnpcmodule.c
  creating build\lib.win32-2.7
  C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for Pytho
n\9.0\VC\Bin\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:c:\python27\libs /LI
BPATH:c:\python27\PCbuild /EXPORT:initminiupnpc build\temp.win32-2.7\Release\min
iupnpcmodule.obj libminiupnpc.a /OUT:build\lib.win32-2.7\miniupnpc.pyd /IMPLIB:b
uild\temp.win32-2.7\Release\miniupnpc.lib /MANIFESTFILE:build\temp.win32-2.7\Rel
ease\miniupnpc.pyd.manifest
  LINK : fatal error LNK1181: cannot open input file 'libminiupnpc.a'
  error: command 'C:\\Users\\gappleto\\AppData\\Local\\Programs\\Common\\Microso
ft\\Visual C++ for Python\\9.0\\VC\\Bin\\link.exe' failed with exit status 1181

  ----------------------------------------
  Failed building wheel for miniupnpc
Failed to build miniupnpc
Installing collected packages: miniupnpc
  Running setup.py install for miniupnpc
    Complete output from command c:\python27\python.exe -c "import setuptools, t
okenize;__file__='c:\\users\\gappleto\\appdata\\local\\temp\\pip-build-wq5lzh\\m
iniupnpc\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read(
).replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\gappleto\
appdata\local\temp\pip-mzwxnc-record\install-record.txt --single-version-externa
lly-managed --compile:
    cc -fPIC -O -Wall -W -Wstrict-prototypes -fno-common -DMINIUPNPC_SET_SOCKET_
TIMEOUT -DMINIUPNPC_GET_SRC_ADDR -D_BSD_SOURCE -D_POSIX_C_SOURCE=1 -ansi   -c -o
 upnpc.o upnpc.c
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, uname -s, ...) failed.
    process_begin: CreateProcess(NULL, cc -fPIC -O -Wall -W -Wstrict-prototypes
-fno-common -DMINIUPNPC_SET_SOCKET_TIMEOUT -DMINIUPNPC_GET_SRC_ADDR -D_BSD_SOURC
E -D_POSIX_C_SOURCE=1 -ansi -c -o upnpc.o upnpc.c, ...) failed.
    make (e=2): The system cannot find the file specified.
    make: *** [upnpc.o] Error 2
    running install
    running build
    running build_ext
    building 'miniupnpc' extension
    C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for Pyt
hon\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include
 -Ic:\python27\PC /Tcminiupnpcmodule.c /Fobuild\temp.win32-2.7\Release\miniupnpc
module.obj
    miniupnpcmodule.c
    C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for Pyt
hon\9.0\VC\Bin\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:c:\python27\libs /
LIBPATH:c:\python27\PCbuild /EXPORT:initminiupnpc build\temp.win32-2.7\Release\m
iniupnpcmodule.obj libminiupnpc.a /OUT:build\lib.win32-2.7\miniupnpc.pyd /IMPLIB
:build\temp.win32-2.7\Release\miniupnpc.lib /MANIFESTFILE:build\temp.win32-2.7\R
elease\miniupnpc.pyd.manifest
    LINK : fatal error LNK1181: cannot open input file 'libminiupnpc.a'
    error: command 'C:\\Users\\gappleto\\AppData\\Local\\Programs\\Common\\Micro
soft\\Visual C++ for Python\\9.0\\VC\\Bin\\link.exe' failed with exit status 118
1

    ----------------------------------------
Command "c:\python27\python.exe -c "import setuptools, tokenize;__file__='c:\\us
ers\\gappleto\\appdata\\local\\temp\\pip-build-wq5lzh\\miniupnpc\\setup.py';exec
(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'),
 __file__, 'exec'))" install --record c:\users\gappleto\appdata\local\temp\pip-m
zwxnc-record\install-record.txt --single-version-externally-managed --compile" f
ailed with error code 1 in c:\users\gappleto\appdata\local\temp\pip-build-wq5lzh
\miniupnpc
```

gappleto97/Senior-Project#3

Output from setupmingw32.py:

```C:\Users\gappleto\Documents\miniupnp-master\miniupnp-master\miniupnpc>python setupmingw32.py build
running build
running build_ext
building 'miniupnpc' extension
creating build
creating build\temp.win32-2.7
creating build\temp.win32-2.7\Release
C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3
 /GS- /DNDEBUG -IC:\Python27\include -IC:\Python27\PC /Tcminiupnpcmodule.c /Fobuild\temp.win32-2.7\Release\miniupnpcmodu
le.obj
miniupnpcmodule.c
creating build\lib.win32-2.7
C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\link.exe /DLL /nologo /INCREM
ENTAL:NO /LIBPATH:C:\Python27\libs /LIBPATH:C:\Python27\PCbuild ws2_32.lib iphlpapi.lib /EXPORT:initminiupnpc build\temp
.win32-2.7\Release\miniupnpcmodule.obj libminiupnpc.a /OUT:build\lib.win32-2.7\miniupnpc.pyd /IMPLIB:build\temp.win32-2.
7\Release\miniupnpc.lib /MANIFESTFILE:build\temp.win32-2.7\Release\miniupnpc.pyd.manifest
LINK : fatal error LNK1181: cannot open input file 'libminiupnpc.a'
error: command 'C:\\Users\\gappleto\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\l
ink.exe' failed with exit status 1181
```

Tried building the solution first (because apparently I'm an idiot and
didn't make the leap that that was necessary), and now I'm getting a
different exit code.

```python
C:\Users\gappleto\Documents\miniupnp-master\miniupnp-master\miniupnpc>python
setupmingw32.py build
running build
running build_ext
building 'miniupnpc' extension
C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for
Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3
 /GS- /DNDEBUG -IC:\Python27\include -IC:\Python27\PC /Tcminiupnpcmodule.c
/Fobuild\temp.win32-2.7\Release\miniupnpcmodu
le.obj
miniupnpcmodule.c
C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for
Python\9.0\VC\Bin\link.exe /DLL /nologo /INCREM
ENTAL:NO /LIBPATH:C:\Python27\libs /LIBPATH:C:\Python27\PCbuild ws2_32.lib
iphlpapi.lib /EXPORT:initminiupnpc build\temp
.win32-2.7\Release\miniupnpcmodule.obj
C:\Users\gappleto\Documents\miniupnp-master\miniupnp-master\miniupnpc\msvc\Releas
e\miniupnpc.lib /OUT:build\lib.win32-2.7\miniupnpc.pyd
/IMPLIB:build\temp.win32-2.7\Release\miniupnpc.lib /MANIFESTFILE:
build\temp.win32-2.7\Release\miniupnpc.pyd.manifest
fatal error C1007: unrecognized flag '-FitObjData' in 'p2'
LINK : fatal error LNK1257: code generation failed
error: command
'C:\\Users\\gappleto\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual
C++ for Python\\9.0\\VC\\Bin\\l
ink.exe' failed with exit status 1257
```

On Tue, Nov 17, 2015 at 1:50 PM, gabe appleton <gappleto97@gmail.com> wrote:

> I know it works under Linux and OSX. I use Linux for my development. I
> also know that it (at least in theory) works on Windows because Bitcoin
> Core uses it.
>
> It would be very useful if you could help me out. If you don't have a copy
> of Windows on hand, you could always use AppVeyor. That's what I've been
> testing my Windows builds on lately.
>
> In the meantime I'll take a look at the links you sent. See if any of
> those changes help.
>
> On Tue, Nov 17, 2015 at 1:00 AM, Thomas BERNARD <notifications@github.com>
> wrote:
>
>> see http://miniupnp.tuxfamily.org/forum/viewtopic.php?t=1815
>> http://miniupnp.tuxfamily.org/forum/viewtopic.php?t=770
>> many people have trouble compiling the python module for windows...
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/miniupnp/miniupnp/issues/159#issuecomment-157312802>.
>>
>
>

Tried building the solution first (because apparently I'm an idiot and
didn't make the leap that that was necessary), and now I'm getting a
different exit code.

```
C:\Users\gappleto\Documents\miniupnp-master\miniupnp-master\miniupnpc>python
setupmingw32.py build
running build
running build_ext
building 'miniupnpc' extension
C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for
Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3
 /GS- /DNDEBUG -IC:\Python27\include -IC:\Python27\PC /Tcminiupnpcmodule.c
/Fobuild\temp.win32-2.7\Release\miniupnpcmodu
le.obj
miniupnpcmodule.c
C:\Users\gappleto\AppData\Local\Programs\Common\Microsoft\Visual C++ for
Python\9.0\VC\Bin\link.exe /DLL /nologo /INCREM
ENTAL:NO /LIBPATH:C:\Python27\libs /LIBPATH:C:\Python27\PCbuild ws2_32.lib
iphlpapi.lib /EXPORT:initminiupnpc build\temp
.win32-2.7\Release\miniupnpcmodule.obj
C:\Users\gappleto\Documents\miniupnp-master\miniupnp-master\miniupnpc\msvc\Releas
e\miniupnpc.lib /OUT:build\lib.win32-2.7\miniupnpc.pyd
/IMPLIB:build\temp.win32-2.7\Release\miniupnpc.lib /MANIFESTFILE:
build\temp.win32-2.7\Release\miniupnpc.pyd.manifest
fatal error C1007: unrecognized flag '-FitObjData' in 'p2'
LINK : fatal error LNK1257: code generation failed
error: command
'C:\\Users\\gappleto\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual
C++ for Python\\9.0\\VC\\Bin\\l
ink.exe' failed with exit status 1257
```

I can't build python module either. Here's what I've tried:
`Microsoft Windows [version 10.0.10586]
(c) 2015 Microsoft Corporation. Tous droits réservés.

C:\miniupnpc-1.9>mingw32-make pythonmodule3
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
cc -fPIC -O -Wall -W -Wstrict-prototypes -fno-common -DMINIUPNPC_SET_SOCKET_TIMEOUT -DMINIUPNPC_GET_SRC_ADDR -D_BSD_SOURCE -D_POSIX_C_SOURCE=1 -ansi   -c -o minissdpc.o minissdpc.c
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, cc -fPIC -O -Wall -W -Wstrict-prototypes -fno-common -DMINIUPNPC_SET_SOCKET_TIMEOUT -DMINIUPNPC_GET_SRC_ADDR -D_BSD_SOURCE -D_POSIX_C_SOURCE=1 -ansi -c -o minissdpc.o minissdpc.c, ...) failed.
make (e=2): Le fichier spécifié est introuvable.
<builtin>: recipe for target 'minissdpc.o' failed
mingw32-make: *** [minissdpc.o] Error 2

C:\miniupnpc-1.9>mingw32-make -f Makefile.mingw pythonmodule
gcc   wingenminiupnpcstrings.o  -lws2_32 -liphlpapi -o wingenminiupnpcstrings
wingenminiupnpcstrings miniupnpcstrings.h.in miniupnpcstrings.h
Windows 6.2 Build 9200
MiniUPnPc version 1.9
15 lines written to miniupnpcstrings.h.
gcc -Wall -Os -DNDEBUG -D_WIN32_WINNT=0X501 -DSTATICLIB -c -o miniwget.o miniwget.c
gcc -Wall -Os -DNDEBUG -D_WIN32_WINNT=0X501 -DMINIUPNP_EXPORTS -c -o dll/miniwget.o miniwget.c
gcc -Wall -Os -DNDEBUG -D_WIN32_WINNT=0X501 -DSTATICLIB -c -o minixml.o minixml.c
gcc -Wall -Os -DNDEBUG -D_WIN32_WINNT=0X501 -DMINIUPNP_EXPORTS -c -o dll/minixml.o minixml.c
gcc -Wall -Os -DNDEBUG -D_WIN32_WINNT=0X501 -DSTATICLIB -c -o minisoap.o minisoap.c
gcc -Wall -Os -DNDEBUG -D_WIN32_WINNT=0X501 -DMINIUPNP_EXPORTS -c -o dll/minisoap.o minisoap.c
ar cr libminiupnpc.a miniwget.o minixml.o minisoap.o
setupmingw32.py build --compiler=mingw32
process_begin: CreateProcess(NULL, python C:\miniupnpc-1.9\setupmingw32.py build --compiler=mingw32, ...) failed.
make (e=2): Le fichier spécifié est introuvable.
Makefile.mingw:36: recipe for target 'pythonmodule' failed
mingw32-make: *** [pythonmodule] Error 2`

C:\miniupnpc-1.9>py setupmingw32.py build
running build
running build_ext
building 'miniupnpc' extension
error: Unable to find vcvarsall.bat

@miniupnp 
I am trying to build the module for Python 3.7 64-bit.

I've managed to build the object files but I have no experience with c. Any chance you can build a 64 bit python module in appveyor?

X:\miniupnp\miniupnpc>python setupmingw32.py build
running build
running build_ext
building 'miniupnpc' extension
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.2
0.27508\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\Users\jo
hn\AppData\Local\Programs\Python\Python37\include -Ic:\Users\john\AppData\Local\
Programs\Python\Python37\include "-IC:\Program Files (x86)\Microsoft Visual Stud
io\2019\Community\VC\Tools\MSVC\14.20.27508\include" "-IC:\Program Files (x86)\W
indows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits
\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\inclu
de\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763
.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinr
t" /Tcminiupnpcmodule.c /Fobuild\temp.win-amd64-3.7\Release\miniupnpcmodule.obj
miniupnpcmodule.c
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.2
0.27508\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EM
BED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\Users\john\AppData\Local\Programs\Python\Py
thon37\libs /LIBPATH:c:\Users\john\AppData\Local\Programs\Python\Python37\PCbuil
d\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\
VC\Tools\MSVC\14.20.27508\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits
\10\lib\10.0.17763.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\
lib\10.0.17763.0\um\x64" ws2_32.lib iphlpapi.lib legacy_stdio_definitions.lib /E
XPORT:PyInit_miniupnpc build\temp.win-amd64-3.7\Release\miniupnpcmodule.obj libm
iniupnpc.a /OUT:build\lib.win-amd64-3.7\miniupnpc.cp37-win_amd64.pyd /IMPLIB:bui
ld\temp.win-amd64-3.7\Release\miniupnpc.cp37-win_amd64.lib
   Creating library build\temp.win-amd64-3.7\Release\miniupnpc.cp37-win_amd64.li
b and object build\temp.win-amd64-3.7\Release\miniupnpc.cp37-win_amd64.exp
miniupnpcmodule.obj : error LNK2001: unresolved external symbol FreeUPNPUrls
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_DeletePortM
apping
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetExternal
IPAddress
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetTotalPac
ketsSent
miniupnpcmodule.obj : error LNK2001: unresolved external symbol freeUPNPDevlist
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetTotalByt
esSent
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetGenericP
ortMappingEntry
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetSpecific
PortMappingEntry
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetValidIGD

miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetTotalPac
ketsReceived
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetTotalByt
esReceived
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_AddAnyPortM
apping
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_DeletePortM
appingRange
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetConnecti
onTypeInfo
miniupnpcmodule.obj : error LNK2001: unresolved external symbol strupnperror
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetStatusIn
fo
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_AddPortMapp
ing
miniupnpcmodule.obj : error LNK2001: unresolved external symbol UPNP_GetPortMapp
ingNumberOfEntries
miniupnpcmodule.obj : error LNK2001: unresolved external symbol upnpDiscover
build\lib.win-amd64-3.7\miniupnpc.cp37-win_amd64.pyd : fatal error LNK1120: 19 u
nresolved externals
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Communit
y\\VC\\Tools\\MSVC\\14.20.27508\\bin\\HostX86\\x64\\link.exe' failed with exit s
tatus 1120

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,19 +1,19 @@
 environment:
   APP_VERSION: '2.1'
   matrix:
-    - PYTHON_VER: C:\Python27\python
+    - PYTHON_VER: C:\Python27
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
       SETUP_COMPILER_FLAG: --compiler=mingw32
       ARCH_BITS: 32
-    - PYTHON_VER: C:\Python35\python
+    - PYTHON_VER: C:\Python35
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
       SETUP_COMPILER_FLAG:
       ARCH_BITS: 32
-    - PYTHON_VER: C:\Python37\python
+    - PYTHON_VER: C:\Python37
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
       SETUP_COMPILER_FLAG:
       ARCH_BITS: 32
-    - PYTHON_VER: C:\Python37-x64\python
+    - PYTHON_VER: C:\Python37-x64
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
       SETUP_COMPILER_FLAG:
       ARCH_BITS: 64
@@ -24,15 +24,18 @@ version: '$(APP_VERSION).{build}'
 image: Visual Studio 2017
 
 install:
-  - set PATH=%PATH%;C:\msys64\mingw%ARCH_BITS%\bin
+  - set PATH=%PYTHON_VER%;%PYTHON_VER%\Scripts;%PATH%;C:\msys64\mingw%ARCH_BITS%\bin
   # We need wheel installed to build wheels
-  - "%PYTHON_VER%.exe -m pip install wheel"
+  - "%PYTHON_VER%\\python.exe -m pip install wheel"
+  - echo %PATH%
+  - cc -v
 
 build_script:
   - cd miniupnpc
   - appveyor AddCompilationMessage "Building miniupnpc"
   - mingw32-make -f Makefile.mingw
-  - mingw32-make -f Makefile.mingw pythonmodule PYTHON=%PYTHON_VER%
+  - appveyor AddCompilationMessage "Building python module"
+  - mingw32-make -f Makefile.mingw pythonmodule PYTHON=%PYTHON_VER%\python
 
 after_build:
   - 7z a -x!wingenminiupnpcstrings.exe ..\miniupnpc_%ARCH_BITS%b-%APPVEYOR_BUILD_VERSION%.zip *.exe *.dll *.a *.lib

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -10,6 +10,9 @@ environment:
     - PYTHON_VER: C:\Python37\python
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
       SETUP_COMPILER_FLAG:
+    - PYTHON_VER: C:\Python37-x64\python
+      APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
+      SETUP_COMPILER_FLAG:
 
 
 version: '$(APP_VERSION).{build}'

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -4,15 +4,19 @@ environment:
     - PYTHON_VER: C:\Python27\python
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
       SETUP_COMPILER_FLAG: --compiler=mingw32
+      ARCH_BITS: 32
     - PYTHON_VER: C:\Python35\python
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
       SETUP_COMPILER_FLAG:
+      ARCH_BITS: 32
     - PYTHON_VER: C:\Python37\python
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
       SETUP_COMPILER_FLAG:
+      ARCH_BITS: 32
     - PYTHON_VER: C:\Python37-x64\python
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
       SETUP_COMPILER_FLAG:
+      ARCH_BITS: 64
 
 
 version: '$(APP_VERSION).{build}'
@@ -20,7 +24,7 @@ version: '$(APP_VERSION).{build}'
 image: Visual Studio 2017
 
 install:
-  - set PATH=%PATH%;C:\msys64\mingw32\bin
+  - set PATH=%PATH%;C:\msys64\mingw%ARCH_BITS%\bin
   # We need wheel installed to build wheels
   - "%PYTHON_VER%.exe -m pip install wheel"
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -35,10 +35,10 @@ build_script:
   - mingw32-make -f Makefile.mingw pythonmodule PYTHON=%PYTHON_VER%
 
 after_build:
-  - 7z a -x!wingenminiupnpcstrings.exe ..\miniupnpc-%APPVEYOR_BUILD_VERSION%.zip *.exe *.dll *.a *.lib
+  - 7z a -x!wingenminiupnpcstrings.exe ..\miniupnpc_%ARCH_BITS%b-%APPVEYOR_BUILD_VERSION%.zip *.exe *.dll *.a *.lib
 
 artifacts:
-  - path: miniupnpc-$(appveyor_build_version).zip
+  - path: miniupnpc*-$(appveyor_build_version).zip
     name: miniupnpc binaries
   - path: miniupnpc/dist/miniupnpc-*.whl
     name: miniupnpc python wheel (self-contained)

